plugins/importexport/medra/MedraExportPlugin.inc.php

```diff
@@ -115,14 +130,62 @@ function depositXML($objects, $context, $filename) {
 				$this->saveRegisteredDoi($context, $object);
 			}
 		} else {
-			// Handle errors.
-			if (is_string($result)) {
-				$result = array(
-					array('plugins.importexport.common.register.error.mdsError', $result)
-				);
-			} else {
-				$result = false;
-			}
+			// Handle errors. There are validations before sending the request of submission to Crossref endpoint, and there is a need to throw a readable exception to the end user
+			//the exception are shown in a table as represented in the code below.
+		    if(!assert(PKPString::regexp_match('#<returnCode>success</returnCode>#', $result))){
```

Comment:
I don't think `assert` is what you really want here. A `false` condition appears to be a valid execution path, but false assertions shouldn't be used in that case.